tool/gravity/cli/reconfigure.go

```diff
@@ -112,28 +112,28 @@ func startReconfiguratorFromService(env *localenv.LocalEnvironment, config Insta
 	if err != nil {
 		return trace.Wrap(utils.NewPreconditionFailedError(err))
 	}
-	installer, err := newReconfigurator(ctx, installerConfig, state)
+	installer, err := newReconfigurator(ctx, *installerConfig, state)
 	if err != nil {
 		return trace.Wrap(utils.NewPreconditionFailedError(err))
 	}
 	interrupt.AddStopper(installer)
 	return trace.Wrap(installer.Run(listener))
 }
 
-func newReconfigurator(ctx context.Context, config *install.Config, state *localenv.LocalState) (*install.Installer, error) {
+func newReconfigurator(ctx context.Context, config install.Config, state localenv.LocalState) (*install.Installer, error) {
 	engine, err := reconfigure.NewEngine(reconfigure.Config{
 		Operator: config.Operator,
-		State:    state,
+		State:    &state,
 	})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	config.LocalAgent = false // To make sure agent does not get launched on this node.
 	installer, err := install.New(ctx, install.RuntimeConfig{
-		Config:         *config,
-		Planner:        reconfigure.NewPlanner(config, state.Cluster),
-		FSMFactory:     reconfigure.NewFSMFactory(*config),
-		ClusterFactory: install.NewClusterFactory(*config),
+		Config:         config,
+		Planner:        reconfigure.NewPlanner(&config, state.Cluster, state.Cluster.ClusterState.Servers[0].Role),
```

Comment:
Be on the safe side and check Servers length?

Reply:
`localState` is validated prior to entering this.